### ISSUE TITLE
How does this compare to MikePopoloski/StringFormatter?

### ISSUE THREAD

**Issue by jamesford42** [Cysharp/ZString#98]
I was using StringBuilder, and just converted to using this:
[MikePopoloski/StringFormatter](https://github.com/MikePopoloski/StringFormatter)

The project is dead, but it does work. I had to jiggle a few things to fully match the same AIP as StringBuilder, but now it does. Someone has added to it (and i have pulled it in) custom format specifiers for numbers, which _might_ have some implementation issues, not sure on that yet.

So..
1. Is anyone even aware of MikePopoloski/StringFormatter or by looking at it say what the likely comparison is to there?
2. How well does this project here work as a 'drop in' StringBuilder replacement?

**Comment by jamesford42** [comment]
Actually I have a better question. So, this project looks like it has gotten a fair amount of use, from people with Unity (which i would also be using this for), and thats good, but it is an older project that cannot be upgraded, and is using 2020.3.32f1. 

My first attempt of pulling the code from this project into my own (just the .csproj not the unity project) shows that it uses C# 8.0 language version features, so that is an issue. I can change the csproj to <LangVersion>8.0</LangVersion> but i suspect that might break things in the unity editor.

**Comment by to11mtm** [comment]
> How well does this project here work as a 'drop in' StringBuilder replacement?

It's pretty close. Some method names are different or you might have to do some things with args but when I tried to do so, the main challenge I ran into was that there was not a built-in `ReplaceAt()` for single characters (and the normal replace did too many bounds checks). Most other things I ran into were easy enough to refactor though.

Cannot speak to the unity side however.
